### PR TITLE
Add GET support, and descriptive output for bad requests/responses

### LINKED ISSUE #6: Fix error with get document by id request
### 💡 Description

When I test this request with the pds-registry-client, I am getting an error:
```
pds-registry-client '/jaxa-registry/_doc/urn:nasa:pds:insight_rad:data_derived::7.0'
Traceback (most recent call last):
  File "/Users/loubrieu/PycharmProjects/serverless-opensearch-request-signer/venv/lib/python3.9/site-packages/requests/models.py", line 974, in json
    return complexjson.loads(self.text, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/loubrieu/PycharmProjects/serverless-opensearch-request-signer/venv/bin/pds-registry-client", line 8, in <module>
    sys.exit(main())
  File "/Users/loubrieu/PycharmProjects/serverless-opensearch-request-signer/src/pds/aossrequestsigner/run.py", line 142, in main
    run(
  File "/Users/loubrieu/PycharmProjects/serverless-opensearch-request-signer/src/pds/aossrequestsigner/run.py", line 62, in run
    output = json.dumps(response.json(), indent=2) if prettify_output else json.dumps(response.json())
  File "/Users/loubrieu/PycharmProjects/serverless-opensearch-request-signer/venv/lib/python3.9/site-packages/requests/models.py", line 978, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```

I am using the opensearch deployed on MCP dev account.

### ⚔️ Parent Epic / Related Tickets

_No response_

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Adds curl-style automatic resolution of request verbs - only GET/POST currently supported.  If --data or the new --matchall option is used, request will be POST, else GET.

--data no longer defaults to a match-all query. --matchall is provided as a shortcut for this common query

Adds url-encoding of path, skippable with new option --noencode

Previously, non-200 responses were only detectable via inspection of the exit code. Now, if `--silent` is not set, a descriptive message will be printed to the console and any response content will be included (ex. for an HTTP400)

## ⚙️ Test Data and/or Report
Tested manually

## ♻️ Related Issues
fixes #6

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -169,7 +169,10 @@ docstring_convention = google
 # • ET Tufte, _Seeing with Fresh Eyes: Meaning, Space, Data, Truth_, Graphics
 #   Press 2020, p.14.
 
-extend-ignore = E203, E501, W503
+#extend-ignore = E203, E501, W503, D415
+#### BEGIN TEMPORARY IGNORE
```
Comment (nutjob4life) [inline setup.cfg:173]: "Temporary" eh? 😏

File: src/pds/aossrequestsigner/errors.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from typing import Optional
+
+
+class Non200HttpStatusError(RuntimeError):
+    def __init__(self, status_code: int, description: Optional[str] = None):
+        if status_code == 200:
```
Comment (nutjob4life) [inline src/pds/aossrequestsigner/errors.py:6]: There's `http.HTTPStatus.OK` that can be used in place of `200` … but given that `200` appears in the name of the class—and "200" is pretty universally understood these days—maybe an explicit `200` is fine

File: src/pds/aossrequestsigner/run.py [hunk 3]
```diff
@@ -42,13 +43,14 @@ def run(
 
     auth = RequestsAWSV4SignerAuth(credentials, aws_region, "aoss")
 
+    request_f = requests.post if data else requests.get
     url = urllib.parse.urljoin(aoss_endpoint, request_path)
     if verbose:
-        print(f"Making request to url: {url}")
+        print(f"Making {'POST' if data else 'GET'} request to url: {url}")
```
Comment (nutjob4life) [inline src/pds/aossrequestsigner/run.py:49]: Is the output of this program ever the input to another (Unix philosophy)? If so, maybe add `file=sys.stderr` to `print()` so verbose diagnostics aren't parsed

**Review by nutjob4life — APPROVED** [review]
Looks fine 👍

File: setup.cfg (hunk 1 above)
Comment (alexdunnjpl) [reply setup.cfg:173]: Hey, take it up with Jordan 😅

File: src/pds/aossrequestsigner/errors.py (hunk 2 above)
Comment (alexdunnjpl) [reply src/pds/aossrequestsigner/errors.py:6]: Better is betterrer - will fix.

File: src/pds/aossrequestsigner/run.py (hunk 3 above)
Comment (alexdunnjpl) [reply src/pds/aossrequestsigner/run.py:49]: Not unforeseeable that someone would like to pipe it to `jq` - will fix

File: src/pds/aossrequestsigner/errors.py (hunk 2 above)
Comment (alexdunnjpl) [reply src/pds/aossrequestsigner/errors.py]: 96b2934c

File: src/pds/aossrequestsigner/run.py (hunk 3 above)
Comment (alexdunnjpl) [reply src/pds/aossrequestsigner/run.py]: ceca1f40

**Pull request merged**